src/VolleyManagement.Backend/VolleyManagement.Services/TournamentService.cs

```diff
@@ -737,31 +738,57 @@ private static int GetGamesCount(int teamsCount)
         private void CreateSchedule(int tournamentId, int allTeamsCount)
         {
             var tournament = Get(tournamentId);
+            var scheduledGames = _gameService.GetTournamentGames(tournamentId)
+                   .Where(tr => tr.Round == FIRST_ROUND);
+
             if (tournament.Scheme == TournamentSchemeEnum.PlayOff
-                && allTeamsCount > DONT_CREATE_SCHEDULE_TEAMS_COUNT)
+                && allTeamsCount > DONT_CREATE_SCHEDULE_TEAMS_COUNT
+                && scheduledGames.Count() != GetGamesCount(allTeamsCount))
```

Comment:
What is th goal of this codition

Reply:
For example: 
when you add 3/4 team to tournament, schedule will be genarated for 4 team,
when you add 5/6/7/8 team to tournament, schedule will be genarated for 8 team ....
Goal is not to generate schedule, if the schedule is already generate for that count of team

I'm working on UTs)

Reply:
I see. I made mistake in condition.

Reply:
Is it enough to check the number of execute remove from _gameRepository?

Reply:
Didn't get last one.

My main concern is that you won't be able to tell me what this line checks in a month time.